agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java

```diff
@@ -184,7 +196,18 @@ public void customize(AutoConfigurationCustomizer autoConfiguration) {
     }
 
     // initialize StatsbeatModule
-    statsbeatModule.start(telemetryClient, configuration);
+    if (telemetryClient.getConnectionString() != null) {
```

Comment:
Can we add a comment on the reason for this check?

Reply:
we don't want to start statsbeat when connection string is null. we have this check in many places.